app/src/main/java/com/example/dine_and_donate/HomeFragments/MapFragment.java

```diff
@@ -420,4 +448,67 @@ private void slideDownMenu() {
         slideView.startAnimation(animate);
     }
 
+    private void setUpCarousel(final DataSnapshot snapshot) {
+        CarouselPicker carouselPicker = (CarouselPicker) slideView.findViewById(R.id.carousel);
+
+        final ArrayList<String> eventIDs = new ArrayList<String>();
+        List<CarouselPicker.PickerItem> mixItems = new ArrayList<>();
+
+        for (DataSnapshot eventChild : snapshot.getChildren()) {
+            mixItems.add(new CarouselPicker.TextItem(getDate((long) eventChild.child("startTime").getValue(),
+                    "MMM dd, yyyy \n hh:mm"),
+                    5));
+            eventIDs.add(eventChild.getKey());
+        }
+
+        CarouselPicker.CarouselViewAdapter mixAdapter = new CarouselPicker.CarouselViewAdapter(slideView.getContext(), mixItems, 0);
+        carouselPicker.setAdapter(mixAdapter);
+
+        btnCreate.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                final Map<String, String> savedEvents = mCurrentUser.getSavedEventsIDs();
+                savedEvents.put(eventIDs.get(position), snapshot.getKey());
+                mRef.child("users").child(mFbUser.getUid()).child("Events").setValue(savedEvents, new DatabaseReference.CompletionListener() {
```

Comment:
not necessary rn, but it's probably worth putting these "users" and "Events" strings into some sort of constants file (like DineAndDonateConstants.java). That way you don't accidentally use the incorrect capitalization/make a typo and need to spend hours debugging. It would be like

public static final String USERS_KEY = "users";
public static final String EVENTS_KEY = "Events";

etc.